src/java/org/apache/cassandra/db/commitlog/CommitLog.java

```diff
@@ -291,7 +291,7 @@ public CommitLogPosition add(Mutation mutation) throws CDCWriteException
                 buffer.putInt((int) checksum.getValue());
 
                 // checksummed mutation
-                dos.write(dob.getData(), 0, size);
+                dos.write(dob.unsafeGetBufferAndFlip());
```

Comment:
I'd revert this, and the similar thing in `UnfilteredSerializer`.

Reply:
why?

Reply:
CommitLog does not hold ref after, `.write` copies the value into its local buffer.  

```
public byte[] getData()
    {
        assert buffer.arrayOffset() == 0;
        return buffer.array();
    }
```

this method is no longer safe as it may be direct now...

Reply:
Makes sense. I was only thinking in terms of whether we copied or not.